internal/pkg/cli/app_status.go

```diff
@@ -0,0 +1,446 @@
+		}
+	} else {
+		app, err := o.storeSvc.GetApplication(o.ProjectName(), o.appName)
+		if err != nil {
+			return fmt.Errorf("get application: %w", err)
+		}
+		appNames = []string{app.Name}
+	}
+
+	if o.envName == "" {
+		envs, err = o.storeSvc.ListEnvironments(o.ProjectName())
+		if err != nil {
+			return fmt.Errorf("list environments: %w", err)
+		}
+		if len(envs) == 0 {
+			return fmt.Errorf("no environments found in project %s", color.HighlightUserInput(o.ProjectName()))
+		}
+	} else {
+		env, err := o.storeSvc.GetEnvironment(o.ProjectName(), o.envName)
+		if err != nil {
+			return fmt.Errorf("get environment: %w", err)
+		}
+		envs = []*archer.Environment{env}
+	}
+
+	appEnvs := make(map[string]appEnv)
+	var appEnvNames []string
+	for _, appName := range appNames {
+		if err := o.initDescriber(o, appName); err != nil {
+			return err
+		}
+		for _, env := range envs {
+			_, err := o.describer.GetServiceArn(env.Name)
+			if err != nil {
+				if isStackNotExistsErr(err) {
+					continue
+				}
+				return fmt.Errorf("check if app %s is deployed in env %s: %w", appName, env.Name, err)
+			}
+			appEnv := appEnv{
+				appName: appName,
+				envName: env.Name,
+			}
+			appEnvName := appEnv.String()
+			appEnvs[appEnvName] = appEnv
+			appEnvNames = append(appEnvNames, appEnvName)
+		}
+	}
+	if len(appEnvNames) == 0 {
+		return fmt.Errorf("no deployed apps found in project %s", color.HighlightUserInput(o.ProjectName()))
+	}
+
+	// return if only one deployed app found
+	if len(appEnvNames) == 1 {
+		log.Infof("Only found one deployed app, defaulting to: %s\n", color.HighlightUserInput(appEnvNames[0]))
+		o.appName = appEnvs[appEnvNames[0]].appName
+		o.envName = appEnvs[appEnvNames[0]].envName
+
+		return nil
+	}
+
+	appEnvName, err := o.prompt.SelectOne(
+		fmt.Sprintf(applicationLogAppNamePrompt),
+		applicationLogAppNameHelpPrompt,
+		appEnvNames,
+	)
+	if err != nil {
+		return fmt.Errorf("select deployed applications for project %s: %w", o.ProjectName(), err)
+	}
+	o.appName = appEnvs[appEnvName].appName
+	o.envName = appEnvs[appEnvName].envName
+
+	return nil
+}
+
+func (o *appStatusOpts) retrieveAllAppNames() ([]string, error) {
+	apps, err := o.storeSvc.ListApplications(o.ProjectName())
+	if err != nil {
+		return nil, fmt.Errorf("list applications for project %s: %w", o.ProjectName(), err)
+	}
+	appNames := make([]string, len(apps))
+	for ind, app := range apps {
+		appNames[ind] = app.Name
+	}
+
+	return appNames, nil
+}
+
+func (o *appStatusOpts) configSvc() error {
+	if err := o.initDescriber(o, o.appName); err != nil {
+		return err
+	}
+	env, err := o.storeSvc.GetEnvironment(o.ProjectName(), o.envName)
+	if err != nil {
+		return fmt.Errorf("get environment: %w", err)
+	}
+	if err := o.initcwSvc(o, env); err != nil {
+		return err
+	}
+	return o.initecsSvc(o, env)
+}
+
+// JSONString returns the stringified webAppStatus struct with json format.
+func (w *webAppStatus) JSONString() (string, error) {
```

Comment:
What do you think about moving `webAppStatus` to the describe package?

We already have structs that implement the same behavior there: JSONString() and HumanString()